trex.org

```diff
@@ -121,23 +121,23 @@
 
 ** Periodic boundary calculations (pbc group)
 
-   A single $k$-point per TREXIO file can be stored. The $k$-point is
-   defined in this group.
 
    #+NAME: pbc
-   | Variable   | Type    | Dimensions | Description             |
-   |------------+---------+------------+-------------------------|
-   | ~periodic~ | ~int~   |            | ~1~: true or ~0~: false |
-   | ~k_point~  | ~float~ | ~(3)~      | $k$-point sampling      |
+   | Variable      | Type    | Dimensions             | Description             |
+   |---------------+---------+------------------------+-------------------------|
+   | ~periodic~    | ~int~   |                        | ~1~: true or ~0~: false |
+   | ~k_point_num~ | ~dim~   |                        | Number of $k$-points    |
+   | ~k_point~     | ~float~ | ~(3, pbc.k_point_num)~ | $k$-point sampling      |
 
    #+CALL: json(data=pbc, title="pbc")
 
    #+RESULTS:
    :results:
    #+begin_src python :tangle trex.json
        "pbc": {
-           "periodic" : [ "int"  , []      ]
-         ,  "k_point" : [ "float", [ "3" ] ]
+              "periodic" : [ "int"  , []                         ]
+         , "k_point_num" : [ "dim"  , []                         ]
+         ,     "k_point" : [ "float", [ "pbc.k_point_num", "3" ] ]
        } ,
    #+end_src
    :end:
```

Comment:
@kousuke-nakano : I have just added the k-vector definitions. Does it look OK?

Reply:
Dear @scemama Thanks! I wonder if the backward compatibility is ok. I stored k point info for each TREXIO file as a list with 3 float numbers e.g., k_point = [/pi, /pi, /pi], but the updated version requires k_point = [[/pi, /pi, /pi]]?

Reply:
Hi @kousuke-nakano ,
it should be OK because the array is flattened in a 1D array. We only need to be careful in the converters because the number of k-points will be missing. Maybe we should write a little script to update your old files to add that there is one k-point.

Reply:
Dear @scemama, thanks for the clarification. Sure, I will make a script to do it if needed.